### PR TITLE
fixed half page scrolls

### LINKED ISSUE #3083: Make <c-u> go to the top of the page
# Is your feature request related to a problem? Please describe.
<c-u> doesn't move the cursor to the top of the file.

# Describe the solution you'd like
move the cursor to the top of the file when it is pressed

# Describe alternatives you've considered
maybe check the length the cursor can go when <c-u> is pressed. When it's less than half a page, just go to the top.

# Additional context
I am typing <c-u> after it is at line 3. It should go to line 1, but it stops
whathappens

### PR DESCRIPTION
**Pull request by PotatosFish**
**What this PR does / why we need it**:
Improves the experience when using <c-u> at the beginning of the page

**Which issue(s) this PR fixes**
Fixes #3083

### CODE REVIEW

**Comment by PotatosFish** [comment]
I have fixed the code so that it actually works, but I don't know where or how to write the tests.

**Comment by PotatosFish** [comment]
never mind I have added two tests for this

File: package-lock.json [hunk 1]
```diff
@@ -1749,8 +1749,7 @@
         "ansi-regex": {
           "version": "2.1.1",
           "bundled": true,
-          "dev": true,
-          "optional": true
+          "dev": true
```
Comment (jpoon) [inline package-lock.json]: Do we need these changes?

File: test/mode/normalModeTests/motions.test.ts [hunk 2]
```diff
@@ -639,4 +639,20 @@ suite('Motions in Normal Mode', () => {
...
+    keysPressed: '<C-u>',
+    end: ['blah', 'duh', 'du|r', 'hur'],
+  });
+
+  /*
+  newTest({
+    title: 'Can handle <C-d> key',
+    start: ['bla|h', 'duh', 'dur', 'hur'],
+    keysPressed: '<C-d>',
+    end: ['blah', 'duh', 'dur', 'hu|r'],
```
Comment (jpoon) [inline test/mode/normalModeTests/motions.test.ts]: Why is this commented out?

Comment (PotatosFish) [reply test/mode/normalModeTests/motions.test.ts]: The test should work but I think the scrolling makes it incorrect

Comment (PotatosFish) [reply test/mode/normalModeTests/motions.test.ts]: I have tried to solve this but cannot make the test pass, so I will leave it commented

Comment (jpoon) [reply test/mode/normalModeTests/motions.test.ts]: Please fix this test.

**Comment by jpoon** [comment]
As there hasn't been any recent activity on this PR, I am closing this out. Feel free to re-open after addressing comments.

**Pull request closed without merging**